enclave/src/test/mocks/attestation_ocall_mock.rs

```diff
@@ -0,0 +1,83 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+	Copyright (C) 2017-2019 Baidu, Inc. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::{ocall::ocall_api::EnclaveAttestationOCallApi, rpc::author::alloc::prelude::v1::Vec};
+use sgx_types::*;
+use std::fmt::{Debug, Formatter, Result as FormatResult};
+
+#[derive(Clone)]
+pub struct AttestationOCallMock {
+	mr_enclave: sgx_measurement_t,
+}
+
+impl AttestationOCallMock {
+	pub fn new() -> Self {
+		Default::default()
```

Comment:
Why do you create this extra level of indirection, and not insert the code of the default implementation here directly?

Reply:
The `Default` is explicitly implemented - so I re-use it here. Or are you questioning the existence of the `new()` method, because any client should use `Default::default()` instead of `AttestationOCallMock::new()` ?

Reply:
I just thought if `new() == default()`, we don't need both.

Reply:
`Default` I need because of a trait bound - but I prefer using an explicit `new()` when creating a mock, because it's not guaranteed that this mock will always implement `Default` (and thus making it more robust)

Reply:
Ok, this reasoning makes sense. Thanks, for clarifying.